complaint_search/tests/test_views_states.py

```diff
@@ -0,0 +1,71 @@
+import copy
+from django.core.urlresolvers import reverse
+
+import mock
+from complaint_search.serializer import SearchInputSerializer
+from complaint_search.defaults import (
+    AGG_EXCLUDE_FIELDS,
+    PARAMS,
+)
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+
+class StatesTests(APITestCase):
+
+    def setUp(self):
+        pass
+
+    def buildDefaultParams(self, overrides):
+        params = copy.deepcopy(PARAMS)
+        params.update(overrides)
+        return params
+
+    @mock.patch('complaint_search.es_interface.states_agg')
+    def test_states_no_param(self, mock_essearch):
+        """
+        Searching with no parameters
+        """
+        url = reverse('complaint_search:states')
+        mock_essearch.return_value = 'OK'
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        mock_essearch.assert_called_once_with(agg_exclude=AGG_EXCLUDE_FIELDS,
+                                              **self.buildDefaultParams({}))
+        self.assertEqual('OK', response.data)
+
+    @mock.patch('complaint_search.es_interface.states_agg')
+    def test_states_with_field__valid(self, mock_essearch):
+        url = reverse('complaint_search:states')
+        for field in SearchInputSerializer.FIELD_CHOICES:
```

Comment:
I like that you are looping through the field choices to test a bunch of scenarios.

It might be cleaner to use the @parameterized tests like we do in CRDB V2 complaints

Reply:
Excellent suggestion!